### ISSUE TITLE
Move Kconfiglib to Zephyr's GitHub org

### ISSUE THREAD

**Issue by carlescufi** [zephyrproject-rtos/zephyr#53894]
## Introduction

As per the conclusions in https://github.com/ulfalizer/Kconfiglib/issues/121, Kconfiglib needs a new home and a new maintainer. @jackrosenthal has volunteered to maintain it, and the Zephyr TSC agreed to host the project under the Zephyr GitHub org.

### Problem description

Given that the pypi package repository contains the [official Kconfiglib packages](https://pypi.org/project/kconfiglib/) and that the original author, @ulfalizer, is the only person with the credentials required to update that package, the only reasonable solution is to fork and rename the library to be able to publish new versions to pypi.

### Proposed change

1. Fork https://github.com/ulfalizer/Kconfiglib into the https://github.com/zephyrproject-rtos org
    - https://github.com/zephyrproject-rtos/Kconfiglib
3. Make all the necessary changes to rename the package, including renaming the repo itself
4. Publish regularly on pypi with the new name, using the Zephyr credentials

### Proposed names

Here's a few names that could be considered:

- `Kconfiglib2`
  - This is my preference because it clearly shows it's a fork of the original `Kconfiglib`, and it also conveys that our intent was not to rename it but we had to
- `Pykconfig`
- `Kconfig`
- `python-kconfig`

**Comment by carlescufi** [comment]
@marc-hb @mbolivar-nordic @tejlmand @nordicjm @stephanosio @fschrempf @gmsotavio

**Comment by nashif** [comment]
one more... 
python-kconfig

**Comment by sjg20** [comment]
Do we need to rename it? Could we take over the project and continue with the same name?

**Comment by carlescufi** [comment]
> > Do we need to rename it? Could we take over the project and continue with the same name?
> 
> It looks like that's something we could request: https://peps.python.org/pep-0541/#how-to-request-a-name-transfer

Let's give this a go. I will create a request today and link it.

**Comment by carlescufi** [comment]
@jackrosenthal here's the request: https://github.com/pypi/support/issues/2526

**Comment by galak** [comment]
pykconfig +1

**Comment by fabiobaltieri** [comment]
`Kconfiglib-ng`? Traditional open source optimistic fork name.

**Comment by marc-hb** [comment]
> python-kconfig

There are 103 Python packages that debian has named by adding a `python-` prefix to their original name: https://packages.debian.org/bullseye/python/

So if `python-kconfig` is ever packaged by Debian they could call `python-python-kconfig`.

Same thing for Fedora: https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_library_naming

Of course Debian and Fedora wouldn't actually do that; I'm merely recommending that Zephyr does not "steal" a very popular naming convention already used by Linux distributions to avoid confusion.

My strong, first preference is to reclaim the existing name if possible, as already requested by @carlescufi. The least disruption by far. I'm optimistic: if this process does not work when the maintainer vanishes then when does it work?

My second preference is `pykconfig`: short, simple and to the point. No hyphen, no mixed case,...

**Comment by sjg20** [comment]
It looks like the requests are being actioned, at least some of them. I added my support into the issue

**Comment by gmsotavio** [comment]
Hello

My thoughts about the subject

Kconfiglib2 - This name sounds fine to me, but I agree with @jackrosenthal that it can be confusing.
Kconfig - May be confused with Linux Kconfig. So it seems like a bad idea.
python-kconfig - That's ok, but this is the default name for Debian packages. So I agree with @marc-hb that if python-kconfig is packaged by Debian they could call it python-python-kconfig. So, I think this name is not optimistic.
pykconfig - The best of the proposed names.

As pointed out by @fabiobaltieri, we could name this project Kconfiglib-ng.

Summing up, +1 vote to Kconfiglib-ng. My second option is pykconfig.

But, for now, it is best to wait for the request sent by @carlescufi to https://peps.python.org/pep-0541/#how-to-request-a-name-transfer

**Comment by jackrosenthal** [comment]
So the name transfer appears to be going nowhere. Anyone opposed to just renaming to `pykconfig`?  Seems like a simple find/replace op.

**Comment by keith-zephyr** [comment]
The feedback we got from one of the pypi maintainers is that requests are handled first-in/first-out.  But because everyone is a volunteer, one should expect a slow response time.

**Comment by marc-hb** [comment]
From
- https://github.com/pypi/support/issues/2526

> Hi all, I will now start the reachability process for your request which can take up to 6 weeks. Thank you for gathering resources regarding the owner, hopefully we can get a response soon and proceed with the transfer.

**Comment by marc-hb** [comment]
https://github.com/pypi/support/issues/2526#issuecomment-2230991181

> Owner role on kconfiglib Project granted to zephyr-project.

**Comment by carlescufi** [comment]
I can confirm that the zephyr-project account in PyPi now manages the Kconfiglib project:
https://pypi.org/project/kconfiglib/
<img width="194" alt="image" src="https://github.com/user-attachments/assets/a1661030-e286-4d73-bcdc-02cb29689fa6">

**Comment by carlescufi** [comment]
@jackrosenthal can you confirm that you are still interested in maintaining Kconfiglib?

**Comment by jackrosenthal** [comment]
Yeah, I'm still available to help maintain kconfiglib!

**Comment by carlescufi** [comment]
Concluding on this issue:

- Repo is now available in https://github.com/zephyrproject-rtos/Kconfiglib (@jackrosenthal has write access)
- PyPi package is now owned by Zephyr

TODO: @jackrosenthal can you comment on all PRs and issues in https://github.com/ulfalizer/Kconfiglib redirecting users to https://github.com/zephyrproject-rtos/Kconfiglib?

**Comment by carlescufi** [comment]
Closing this issue as this is effectively complete.

**Comment by carlescufi** [comment]
Reopening because I now realize we have some commits in Zephyr's own kconfiglib copy (yes, we don't use the PyPi package) that would need to be backported to the new upstream:
https://github.com/zephyrproject-rtos/zephyr/commits/main/scripts/kconfig/kconfiglib.py

**Comment by carlescufi** [comment]
Now complete.

**Comment by marc-hb** [comment]
BTW many "Links" on the left side of https://pypi.org/project/kconfiglib/ still point at https://github.com/ulfalizer/Kconfiglib :-(

Reported in https://github.com/zephyrproject-rtos/Kconfiglib/issues/16
